### PR TITLE
Do 504/spark ci

### PR DESCRIPTION
**Pull request by joseph-tatari**
The `spark` image needs the `dd-spark-metrics` jar in its build.

I first tried a step in the pipeline to use the `amazon/aws-cli` image to download the jar from S3, copy into a temp volume, and then `COPY` that jar in the spark dockerfile.

But, because its being built in docker-in-docker, i had trouble accessing that file from the host filesystem with `ADD`.

Instead, i'm now adding a build step to use the dd-spark-metrics docker image, and then `COPY` the jar from inside that image.

Related PRs will add the `dd-spark-metrics` ecr repo in terraform, and update the `dd-spark-metrics` Drone pipeline to upload the Docker image to ECR instead of the JAR to S3

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -59,3 +65,48 @@ steps:
...
+    commands:
+      - echo "building $SPARK_VERSION"
+      - docker build -t spark:${DRONE_COMMIT_SHA} $SPARK_VERSION
+
+  - name: publish-spark-ecr
+    image: plugins/ecr:19.03
+    when:
+      branch:
+        - master
+        - main
```
Comment (pshelby) [inline .drone.yml:97]: Nice!

**Review by pshelby — APPROVED** [review]
🚀

**Comment by calvintatari** [comment]
Not a bad solution.  It might be worth exploring (at some point) trying the aws-cli route to generate a signed url and then pass that in as a build arg?

**Comment by henryzhangsta** [comment]
@calvintatari Yeah we are eventually going to need to be able to pull data files into build steps, either by granting the drone builder the ability to pull in artifacts, or using the aws-cli. might be worth investigating. @joseph-tatari Where does this jar come from? Is it built by us or off the internet?

**Comment by joseph-tatari** [comment]
> @calvintatari Yeah we are eventually going to need to be able to pull data files into build steps, either by granting the drone builder the ability to pull in artifacts, or using the aws-cli. might be worth investigating. @joseph-tatari Where does this jar come from? Is it built by us or off the internet?

@henryzhangsta we build it here: https://github.com/tatari-tv/dd-spark-metrics/pull/2

**Comment by henryzhangsta** [comment]
I can't see that repo. Can we apply permissions to all repos by default? @escote-tatari

**Review by escote-tatari — APPROVED** [review]

**Review by henryzhangsta — APPROVED** [review]

**Pull request merged**